### PR TITLE
Date formatting

### PR DESCRIPTION
**Pull request by jkbits1**
Resolves https://github.com/voteamerica/voteamerica.github.io/issues/81

Note: each time format has a final letter appended. This is apparently part of the ISO format and represents time zone.

Added moment.js (via cdn) for date formatting

Code needs refactoring but is sound in creation of iso times, storage within input element and creation of string sent to db

### CODE REVIEW

**Comment by dmilet** [comment]
I took the index.html from this pull request and submitted a Rider
The  date/time intervals are not correct
Also, it should be populated in AvailableRideTimesJSON for the node.js app to store it in DB

rider payload: {
    "_redirect": "null/home/david/carpool/voteamerica.github.io/index-john.htmlthanks.html",
    **"hiddenJSONTimes": "",
    "RiderDate": "Thursday, Oct 20",
    "datetime[]": [
        "Thu Oct 13 2016 18:00:46 GMT+0800 (MYT) 9:20am/Thu Oct 13 2016 18:00:46 GMT+0800 (MYT) 6:10pm",
        "Thursday, Oct 20 11:20am/Thursday, Oct 20 3:50pm"
    ],
    "AvailableRideTimesJSON": "{}",**
    "RiderVotingState": "VA",
    "RiderCollectionZIP": "20111",
    "RiderDropOffZIP": "20112",
    "TotalPartySize": "10",
    "NeedWheelchair": "on",
    "TwoWayTripNeeded": "on",
    "RiderFirstName": "David",
    "RiderLastName": "Milet",
    "RiderEmail": "david.milet@gmail.com",
    "RiderPhone": "555-555-5555",
    "RiderIsVulnerable": "on",
    "RiderLegalConsent": "on",
    "RiderWillNotTalkPolitics": "on",
    "PleaseStayInTouch": "on"
}

File: index.html [hunk 1]
```diff
@@ -662,62 +666,78 @@ <h3 class="bannerbox__title">Newsletter</h3>
...
+
+                    //     outputDateTime.startDttm = startDateTimeISO;
+                    //     outputDateTime.endDttm = endDateTimeISO;
+
+                    //     console.log("Date Time pair:" + JSON.stringify(outputDateTime));
+
+                    // } catch (err) {
+                    //     console.error(err);
+                    // }
+                    // availTimesObjectArray.push(outputDateTime);
```
Comment (richardwestenra) [inline index.html]: No need to keep this code commented if it's in the Git history, so we can just delete it, yeah?

File: index.html [hunk 2]
```diff
@@ -25,6 +25,8 @@
     <link type="text/css" rel="stylesheet" href="res/nouislider.min.css">
     <link type="text/css" rel="stylesheet" href="res/styles.css">
 
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/moment.js/2.15.1/moment.min.js"></script>
```
Comment (richardwestenra) [inline index.html]: Can we add this in the footer instead of the header please?

Comment (richardwestenra) [inline index.html]: Also, we now have two separate date/time formatting libraries included: Moment.js and jquery-dateFormat.js (on line 611). I don't mind which we use: Moment.min is ~20kb vs ~5kb for dateFormat, but the former is more widely used ¯\\_(ツ)_/¯. However we should choose one and stick with it for all date formatting use cases, ideally.

File: index.html [hunk 3]
```diff
@@ -872,6 +934,30 @@ <h3 class="bannerbox__title">Newsletter</h3>
 
             return hour + ':' + out_minutes + ampmString;
         }
+
+        function formatTime2(minutes) {
```
Comment (richardwestenra) [inline index.html]: Do we still need both these formatTime functions, or can we get rid of one of them? If we need both, maybe we should name them more descriptively.

File: index.html (hunk 2 above)
Comment (jkbits1) [inline index.html]: I don't have strong views. Really this should be in the node app but we need something working asap and refactor later

File: index.html (hunk 3 above)
Comment (jkbits1) [inline index.html]: It's a refactor thing. we need the new one, maybe not the first. I don't know

**Comment by jkbits1** [comment]
@dmilet this PR resolves the issue for driver available dates. If it works there, it will be applied to rider dates too

**Pull request merged**